src/codegen/compile.ml

```diff
@@ -4548,7 +4577,12 @@ module Text = struct
     set_blob ^^
     get_blob ^^ Blob.as_ptr_len env ^^
     E.call_import env "rts" "utf8_valid" ^^
-    G.if1 I32Type (Opt.inject_simple env get_blob) (Opt.null_lit env)
+    G.if1 I32Type
+      (get_blob ^^ Blob.as_ptr_len env ^^
+       of_ptr_size env ^^ (* creates text blob *)
```

Comment:
Good that you covered this case: We indeed need to create a copy of the blob, as it has a different purpose (and tag)...